src/app/store/status/slice.ts

```diff
@@ -125,8 +126,14 @@ const statusSlice = createSlice({
         // No state changes need to be handled for this action.
       },
     },
-    websocketDisconnected: (state: StatusState) => {
+    websocketDisconnected: (
+      state: StatusState,
+      action: PayloadAction<StatusState["sessionExpired"]>
+    ) => {
       state.connected = false;
+      if (action.payload === true) {
+        state.authenticated = false;
+      }
```

Comment:
As discussed during our session, let's move this to `websocketDisconnect` action.